oak-run/src/main/java/org/apache/jackrabbit/oak/run/FrozenNodeRefsByScanningCommand.java

```diff
@@ -0,0 +1,293 @@
+            Query q = qm.createQuery("SELECT * FROM [nt:base] AS p WHERE PROPERTY(*, '*') LIKE \"________-____-____-____-____________\"", "JCR-SQL2");
+            QueryResult qr = q.execute();
+            NodeIterator it = qr.getNodes();
+
+            while (it.hasNext()) {
+                Node n = it.nextNode();
+                PropertyIterator pit = n.getProperties();
+                while (pit.hasNext()) {
+                    Property p = pit.nextProperty();
+                    if ("jcr:uuid".equals(p.getName())) {
+                        // jcr:uuid should be skipped as that's the identifier of a node, not a reference
+                        continue;
+                    }
+                    if (!p.isMultiple()) {
+                        String propValue = p.getValue().getString();
+                        if (propValue.matches("........-....-....-....-............")) {
+                            verify(list, session, n, p, propValue);
+                        }
+                    } else {
+                        for (Value v : p.getValues()) {
+                            String propValue = v.getString();
+                            if (propValue.matches("........-....-....-....-............")) {
+                                verify(list, session, n, p, propValue);
+                            }
+                        }
+                    }
+                }
+            }
+
+            System.out.println("logout...");
+            session.logout();
+            System.out.println("done.");
+
+            return !list.isEmpty();
+        } catch (Throwable e) {
+            throw closer.rethrow(e);
+        } finally {
+            closer.close();
+        }
+    }
+
+    private void verify(List<FrozenNodeRef> list, Session session, Node n, Property p, String propValue) throws RepositoryException {
+        try {
+            Node node = session.getNodeByIdentifier(propValue);
+            String path = node.getPath();
+            boolean candidate = FrozenNodeRef.isFrozenNodeReferenceCandidate(path);
+            if (!candidate) {
+                return;
+            }
+            Property primaryType = node.getProperty("jcr:primaryType");
+            String primaryTypeValue = primaryType.getString();
+            boolean isNtFrozenNode = "nt:frozenNode".equals(primaryTypeValue);
+            if (!isNtFrozenNode) {
+                // this is where we ultimately have to continue out in any case - as only an nt:frozenNode
+                // is what we're interested in.
+                return;
+            }
+
+            String uuid = propValue;
+            String referrerPath = n.getPath();
+            String referrerProperty = p.getName();
+            FrozenNodeRef ref = new FrozenNodeRef(referrerPath, referrerProperty, PropertyType.nameFromValue(p.getType()), uuid, path);
+            list.add(ref);
```

Comment:
Won't this consume too much memory when there are many references? The only use of the list seems to be to decide on the return of `uuidscan()`.

Reply:
Good point. The list was more for reporting each find - but ultimately there's no other use case for this atm, and the reporting itself can done as it happens. So I've now changed this to only carry a `count` at https://github.com/apache/jackrabbit-oak/pull/238/commits/fb8b330c4f248078bb2dce62cfb19ae90a50030c